Sources/Kafka/RDKafka/RDKafkaClient.swift

```diff
@@ -19,37 +19,51 @@ import Logging
 /// Base class for ``KafkaProducer`` and ``KafkaConsumer``,
 /// which is used to handle the connection to the Kafka ecosystem.
 final class RDKafkaClient: Sendable {
+    // Default size for Strings returned from C API
+    static let stringSize = 1024
+
     /// Determines if client is a producer or a consumer.
     enum ClientType {
         case producer
         case consumer
     }
 
-    // Default size for Strings returned from C API
-    static let stringSize = 1024
-
     /// Handle for the C library's Kafka instance.
     private let kafkaHandle: OpaquePointer
     /// A logger.
     private let logger: Logger
 
-    /// `librdkafka`'s main `rd_kafka_queue_t`.
-    private let mainQueue: OpaquePointer
+    /// `librdkafka`'s `rd_kafka_queue_t` that events are received on.
+    private let queue: OpaquePointer
 
     // Use factory method to initialize
     private init(
+        type: ClientType,
         kafkaHandle: OpaquePointer,
         logger: Logger
     ) {
         self.kafkaHandle = kafkaHandle
         self.logger = logger
 
-        self.mainQueue = rd_kafka_queue_get_main(self.kafkaHandle)
+        if type == .consumer {
+            if let consumerQueue = rd_kafka_queue_get_consumer(self.kafkaHandle) {
+                // (Important)
+                // Polling the queue counts as a consumer poll, and will reset the timer for `max.poll.interval.ms`.
+                self.queue = consumerQueue
+            } else {
+                // Getting consumer queue failed, use main queue.
+                self.queue = rd_kafka_queue_get_main(self.kafkaHandle)
```

Comment:
should we throw an exception in that case?

Reply:
I changed it to a `fatalError` now. Reaching this statement clearly indicates that no `group.id` has been set (which we should do in either case, see link in code) which indicates a programming error.

Thanks for pointing that out!